src/main/java/org/opensrp/connector/openmrs/service/OpenmrsLocationService.java

```diff
@@ -92,7 +94,7 @@ private Location makeLocation(String locationJson) throws JSONException {
 			location.addTag(tags.getJSONObject(i).getString(ConnectorConstants.DISPLAY));
 		}
 
-		if(locationsJsonObject.has(ConnectorConstants.ATTRIBUTES)) {
+		if(addLocationAttributesToPayload && locationsJsonObject.has(ConnectorConstants.ATTRIBUTES)) {
```

Comment:
I don't think this is required, since this processes if the location json from openmrs has the attribute attibutes.

This is the place that the location attributes should be removed for BA custom endpoint 

```
	public List<Location> getAllLocations(List<Location> locationList, int startIndex) throws JSONException {
		String response = this.getURL(HttpUtil.removeEndingSlash(this.OPENMRS_BASE_URL) + "/" + LOCATION_URL +
				"?v=custom:(uuid,display,name,tags:(uuid,display),parentLocation:(uuid,display),attributes)&limit=100&startIndex="+startIndex);
```

The attributes should be removed from custom format

Reply:
You mean create a new and separate endpoint for BA only with the attributes part removed?

Reply:
No `getAllLocations` is called by the BA endpoint, if you remove attributes as on the comment above the code block will not be executed. Retrieving attributes is unnecessary since they are not used in output or determining the output

Reply:
Also the endpoint I fixed does not use the above filters. We'll wan't to add the above but for the other one do we revert that? 

Here's the request `.../openmrs/ws/rest/v1/location/<uuid>?v=full`

It has no custom attributes

Reply:
When testing retrieving attributes added 2 seconds for each request to fetch locations, it also adds to size of request and its not being used

Reply:
there may be some android clients who may be using attributes in the mean time, so we should not change that. Attributes are returned the security/authenticate endpoint which still uses this method

Reply:
gotchu